### PR TITLE
New Contributing Guide

### PR DESCRIPTION
**Pull request by psiinon**
Significant set of changes here.
This is intended to replace things like 

- https://github.com/zaproxy/zaproxy/blob/main/CONTRIBUTING.md
- https://github.com/zaproxy/zaproxy/wiki/DevGuidelines

Question: Should the page titles stay as "Likes" etc or should they be: "Contribute: Likes" etc or ??

Note that I've revampted the top level Documentation page and moved the Dev Guidelines from the wiki pretty much as is (it doesnt look great, but we can improve it over time)

Screenshot 2022-01-20 at 11-48-17 OWASP ZAP – Documentation

Screenshot 2022-01-20 at 11-48-32 OWASP ZAP – Contributing Guide

Screenshot 2022-01-20 at 11-48-42 OWASP ZAP – Likes

Screenshot 2022-01-20 at 11-48-52 OWASP ZAP – Forums

Screenshot 2022-01-20 at 11-58-33 OWASP ZAP – Issues

Screenshot 2022-01-20 at 11-49-24 OWASP ZAP – Docs

Screenshot 2022-01-20 at 11-50-37 OWASP ZAP – Translate

Screenshot 2022-01-20 at 11-50-53 OWASP ZAP – Coding

The Dev Rules and Guideleines are under the Developer Guide section of course ;)

Screenshot 2022-01-20 at 12-00-04 OWASP ZAP – Development Rules and Guidelines

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
> Question: Should the page titles stay as "Likes" etc or should they be: "Contribute: Likes" etc or ??

I’d say “Contribute: Likes”.

**Comment by psiinon** [comment]
I've updated all of the titles to be like "Contribute: Likes" :)

**Comment by psiinon** [comment]
Ooops - didnt include the new Dev Rules page, now added - I do plan to revamp this, when will depend on reviews, it could be in a future PR?

**Comment by kingthorin** [comment]
Further work in future PR is fine with me.

File: site/content/docs/contribute/coding.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+cascade:
+  EditableContent: true
+  addBreadcrumbs: true
+---
+There's always lots of coding to be done!
+
+This can be one of the most rewarding ways to contribute to ZAP - just look at the 
+ZAP [Student Hall of Fame](/student-hall-of-fame/) to see how many students have made major contributions to ZAP and where its led them.
+
+You don't have to be a student of course, anyone who knows either Java or JavaScript will be able to help us out with the ZAP code base.
```
Comment (kingthorin) [inline site/content/docs/contribute/coding.md:13]: I'd say anyone who is interested.

I barely knew Java when I started.

File: site/content/docs/contribute/coding.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+### Getting Started
+
+You should start by becoming familiar with the ZAP Desktop app even if you plan to just help with automation - its very difficult to improve something that you do not understand and the desktop is the best way to learn about the way ZAP works.
```
Comment (kingthorin) [inline site/content/docs/contribute/coding.md:17]:
```suggestion
You should start by becoming familiar with the ZAP Desktop app even if you plan to just help with automation - it's very difficult to improve something that you do not understand and the desktop is the best way to learn about the way ZAP works.
```

File: site/content/docs/contribute/coding.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+Scan rules are a good place to start as they are relatively self contained and do not require a great understanding of the rest of the ZAP codebase.
+
+If you do not have a specific rule you would like to add / improve (and if you do not find a relevant issue that appeals to you) then try running ZAP against one of the many [deliberately vulnerable web applications](https://owasp.org/www-project-vulnerable-web-applications-directory/).
+
+You can see the list of current ZAP rules on the [Alert Details](/docs/alerts/) page.
+
+The following blog posts will help you get started with san rules - they are a bit old but they are mostly still accurate:
+* [Hacking ZAP #3 - Passive scan rules](/blog/2014-04-03-hacking-zap-3-passive-scan-rules/)
+* [Hacking ZAP #4 - Active scan rules](https://www.zaproxy.org/blog/2014-04-30-hacking-zap-4-active-scan-rules/)
```
Comment (kingthorin) [inline site/content/docs/contribute/coding.md:43]: Any reason one has host and one doesn't?

File: site/content/docs/contribute/issues.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title: "Contribute: Issues"
+type: page
+cascade:
+  EditableContent: true
+  addBreadcrumbs: true
+---
+If you are having problems with ZAP then its always best to first look at the FAQ: [Somethings not working. What should I do?](/faq/somethings-not-working-what-should-i-do/).
```
Comment (kingthorin) [inline site/content/docs/contribute/issues.md:8]:
```suggestion
If you are having problems with ZAP then it's always best to first look at the FAQ: [Somethings not working. What should I do?](/faq/somethings-not-working-what-should-i-do/).
```

**Comment by kingthorin** [comment]
I'm sure most of those were issues in existing content, but may as well tiddy them up 👍

File: site/content/docs/contribute/coding.md (hunk 3 above)
Comment (psiinon) [reply site/content/docs/contribute/coding.md:43]: Er, mistake? ;)

**Comment by psiinon** [comment]
Thanks - think most of those were introduced by me this time ;)
Oh, and I couldnt resist tweaking the Dev Rules page:
Screenshot 2022-01-20 at 13-48-46 OWASP ZAP – Development Rules and Guidelines

**Review by kingthorin — APPROVED** [review]

**Comment by psiinon** [comment]
A couple of tweeks thanks to feedback.
Main one is renaming the "Contribute: Likes" page to "Contribute: Social Media".
I've also added the line: "We also love seeing articles about ZAP on other sites, just don’t forget to tell us about them!"

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by sgerlach** [comment]
This looks great as a re-organization and additional details to the Contrubution/Code Guidelines. Excited for more details around "What to expect when making a pull request" and Details around Scan Rule Abstraction (what they do) to help first timers!

**Comment by psiinon** [comment]
Yeah, they are definitely on the list!

**Pull request merged**